includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -667,6 +672,32 @@ private function has_used_attributes( $attribute_names ) {
 
 				$this->used_attributes[ $attribute_name ] = ( 0 !== $this->xpath->query( $expression )->length );
 			}
+
+			// Attributes for amp-accordion, see <https://amp.dev/documentation/components/amp-accordion/#styling>.
```

Comment:
We'll probably have more coming later on, so it might be better to write this as a mapping array with generic logic:
```php
	private function has_used_attributes( $attribute_names ) {
		$dynamically_added_attributes = [
			// Attributes for amp-accordion, see <https://amp.dev/documentation/components/amp-accordion/#styling>.
			'expanded'       => [ 'amp-accordion' ],
			// Attributes for amp-sidebar, see <https://amp.dev/documentation/components/amp-sidebar/#styling>.
			// The 'open' attribute is also used by the HTML5 <details> attribute.
			'open'           => [ 'amp-sidebar', 'details' ],
			// Attributes for amp-live-list, see <https://amp.dev/documentation/components/amp-live-list/#styling>.
			'data-tombstone' => [ 'amp-live-list' ],
		];

		foreach ( $attribute_names as $attribute_name ) {

			// [...]

			foreach ( $dynamically_added_attributes as $attribute => $tags ) {
				if ( $attribute !== $attribute_name ) {
					continue;
				}

				foreach ( $tags as $tag ) {
					if ( $this->has_used_tag_names( [ $tag ] ) ) {
						// This is a dynamically added attribute, so skip removal.
						continue 3;
					}
				}

				return false;
			}

			// [...]
		}
		return true;
	}
```
The list with the mappings can even be extracted into a place that is more manageable as needed.